### PR TITLE
enforce wheel size limits, README formatting in CI

### PR DESCRIPTION
**Pull request by jameslamb**
## Description

Contributes to https://github.com/rapidsai/build-planning/issues/110

Proposes adding 2 types of validation on wheels in CI, to ensure we continue to produce wheels that are suitable for PyPI.

* checks on wheel size (compressed),
  - *to be sure they're under PyPI limits*
  - *and to prompt discussion on PRs that significantly increase wheel sizes*
* checks on README formatting
  - *to ensure they'll render properly as the PyPI project homepages*
  - *e.g. like how https://github.com/scikit-learn/scikit-learn/blob/main/README.rst becomes https://pypi.org/project/scikit-learn/*

### How I tested this

Initially set the size threshold for `libcudf` to a value that I knew it'd violate (75MB compressed, when the wheels are 400+ MB compressed).

Saw CI fail as expected, and print a summary with the expected contents.

```text
checking 'final_dist/libcudf_cu11-24.12.0a333-py3-none-manylinux_2_28_aarch64.whl'
----- package inspection summary -----
file size
  * compressed size: 0.4G
  * uncompressed size: 0.6G
  * compression space saving: 34.6%
contents
  * directories: 164
  * files: 1974 (2 compiled)
size by extension
  * .so - 0.6G (97.0%)
  * .h - 6.7M (1.0%)
  * no-extension - 4.8M (0.7%)
  * .cuh - 3.8M (0.6%)
  * .hpp - 2.2M (0.3%)
  * .a - 1.1M (0.2%)
  * .inl - 0.8M (0.1%)
  * .cmake - 0.1M (0.0%)
  * .md - 8.3K (0.0%)
  * .py - 4.0K (0.0%)
  * .pc - 0.2K (0.0%)
  * .txt - 34.0B (0.0%)
largest files
  * (0.6G) libcudf/lib64/libcudf.so
  * (3.3M) libcudf/bin/flatc
  * (1.0M) libcudf/lib64/libflatbuffers.a
  * (0.5M) libcudf/include/libcudf/rapids/libcudacxx/cuda/std/__atomic/functions/cuda_ptx_generated.h
  * (0.2M) libcudf_cu11-24.12.0a333.dist-info/RECORD
------------ check results -----------
1. [distro-too-large-compressed] Compressed size 0.4G is larger than the allowed size (75.0M).
errors found while checking: 1
```

([build link](https://github.com/rapidsai/cudf/actions/runs/11748370606/job/32732391718?pr=17284#step:13:3062))

Updated that threshold in `python/libcudf/pyproject.toml`, and saw the build succeed (but the summary still printed).

### Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: ci/validate_wheel.sh
Comment (bdice) [inline ci/validate_wheel.sh:1]: Should this be implemented in https://github.com/rapidsai/gha-tools?

Comment (bdice) [reply ci/validate_wheel.sh:1]: Discussed offline -- maybe we can centralize this in the future. There may be some differences in how paths / current directories are handled across repositories. For now this is not a priority.

File: ci/validate_wheel.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+wheel_dir_relative_path=$2
+
+# TODO: pre-install these in ci-imgs
+rapids-logger "installing validation tools"
+python -m pip install \
+    -v \
+    'pydistcheck==0.8.0' \
+    'twine>=5.0.0'
+
+pyenv rehash
```
Comment (jameslamb) [inline ci/validate_wheel.sh:16]: TODO before merging: remove this after `pydistcheck` is added to ci-imgs: https://github.com/rapidsai/ci-imgs/pull/205

(`twine` was already in there)

Comment (jameslamb) [reply ci/validate_wheel.sh]: Merged that PR, saw all the new images published ([build link](https://github.com/rapidsai/ci-imgs/actions/runs/11804580918)), just pushed https://github.com/rapidsai/cudf/pull/17284/commits/4aa1d3c07e4492f02ff34b1365c7d2955e959df6 here removing this install.

**Review by bdice — APPROVED** [review]

**Pull request merged**